### PR TITLE
[dev] statically link libstdc++ and libgcc

### PR DESCRIPTION
**Pull request by Rhett-Ying**
## Description

Old
```
-rwxrwxr-x 1 ubuntu ubuntu  40M Jul 22 08:23 build/dgl_sparse/libdgl_sparse_pytorch_2.2.0.so
-rwxrwxr-x 1 ubuntu ubuntu 107M Jul 22 08:24 build/graphbolt/libgraphbolt_pytorch_2.2.0.so
-rwxrwxr-x 1 ubuntu ubuntu 246M Jul 22 08:23 build/libdgl.so
```

New
```
-rwxrwxr-x 1 ubuntu ubuntu  41M Jul 22 08:15 build/dgl_sparse/libdgl_sparse_pytorch_2.2.0.so
-rwxrwxr-x 1 ubuntu ubuntu 107M Jul 22 08:16 build/graphbolt/libgraphbolt_pytorch_2.2.0.so
-rwxrwxr-x 1 ubuntu ubuntu 248M Jul 22 08:14 build/libdgl.so
```
## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [ ] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

**Review by frozenbugs — APPROVED** [review]

File: CMakeLists.txt [hunk 1]
```diff
@@ -118,6 +118,12 @@ endif()
 
 set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${CMAKE_CURRENT_SOURCE_DIR}/cmake/modules)
 
+# Always statically link libstdc++ and libgcc for portability
+if (NOT MSVC)
```
Comment (mfbalin) [inline CMakeLists.txt:122]: Would this break compatability with clang if anyone is using libc++?

Comment (mfbalin) [reply CMakeLists.txt:122]: https://stackoverflow.com/a/78242553/2904531

Comment (mfbalin) [reply CMakeLists.txt:122]: Looks like there is a way to detech if `libstdc++` is in use.

Comment (Rhett-Ying) [reply CMakeLists.txt]: seems it works for clang as well. but I didn't verify it. so I add gnu check for now.

**Comment by frozenbugs** [comment]
@dgl-bot

**Pull request closed without merging**